tests/sample_size/test_sample_size_calculator.py

```diff
@@ -81,83 +82,82 @@ def test_get_single_sample_size_ttest(self, mock_solve_power):
 
         self.assertEqual(sample_size, test_sample_size)
         mock_get_single_sample_size.assert_called_once()
-        self.assertIsInstance(mock_get_single_sample_size.call_args[0][0], BooleanMetric)
-        assert_equal(mock_get_single_sample_size.call_args[0][0].probability, test_probability)
-        assert_equal(mock_get_single_sample_size.call_args[0][0].mde, test_mde)
+        mock_get_multiple_sample_size.assert_not_called()
+        self.assertIsInstance(mock_get_single_sample_size.call_args[0][0], metric_class)
+        # assert_equal(mock_get_single_sample_size.call_args[0][0].probability, test_probability)
+        self.assertEqual(mock_get_single_sample_size.call_args[0][0].mde, test_mde)
 
+    @patch("sample_size.sample_size_calculator.SampleSizeCalculator.get_multiple_sample_size")
     @patch("sample_size.sample_size_calculator.SampleSizeCalculator._get_single_sample_size")
-    def test_get_overall_sample_size_numeric(self, mock_get_single_sample_size):
-        test_metric_type = "numeric"
-        test_sample_size = 2000
-        mock_get_single_sample_size.return_value = test_sample_size
-
-        test_mde = 5
-        test_variance = 500
-        test_metric_metadata = {"variance": test_variance, "mde": test_mde}
-        calculator = SampleSizeCalculator()
-        calculator.register_metrics([{"metric_type": test_metric_type, "metric_metadata": test_metric_metadata}])
-
-        sample_size = calculator.get_sample_size()
-
-        self.assertEqual(sample_size, test_sample_size)
-        mock_get_single_sample_size.assert_called_once()
-        self.assertIsInstance(mock_get_single_sample_size.call_args[0][0], NumericMetric)
-        assert_equal(mock_get_single_sample_size.call_args[0][0].variance, test_variance)
-        assert_equal(mock_get_single_sample_size.call_args[0][0].mde, test_mde)
-
-    @patch("sample_size.sample_size_calculator.SampleSizeCalculator._get_single_sample_size")
-    def test_get_overall_sample_size_ratio(self, mock_get_single_sample_size):
-        test_metric_type = "ratio"
+    def test_get_overall_sample_size_multiple(self, mock_get_single_sample_size, mock_get_multiple_sample_size):
+        test_metric_type = "boolean"
         test_sample_size = 2000
+        mock_get_multiple_sample_size.return_value = test_sample_size
         mock_get_single_sample_size.return_value = test_sample_size
 
-        test_numerator_mean = 2000
-        test_numerator_variance = 100000
-        test_denominator_mean = 200
-        test_denominator_variance = 2000
-        test_covariance = 5000
-        test_mde = 10
-
-        test_metric_metadata = {
-            "numerator_mean": test_numerator_mean,
-            "numerator_variance": test_numerator_variance,
-            "denominator_mean": test_denominator_mean,
-            "denominator_variance": test_denominator_variance,
-            "covariance": test_covariance,
-            "mde": test_mde,
-        }
-
+        # test_mde = 0.02
```

Comment:
I'm trying to add these back https://github.com/godaddy/sample-size/pull/32#discussion_r889166314. Or are we good to actually skip the metadata check for now? cc @pchiang-godaddy @vjohnson1-godaddy

Reply:
I think you can get rid of this it now because it's capture in the metadata that you parameterized

Reply:
since you actually validate the metric by `calculator.metrics[0]` i'm fine skipping them. but can you add one more check for call count? since `has_calls` doesn't require you to provide all the calls
```python
self.assertEqual(mock_get_single_sample_size.call_count, 4)
```